api/v1alpha2/bundledeployment_types.go

```diff
@@ -52,40 +52,29 @@ type BundleDeploymentSpec struct {
 	//+kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleDeployment.
 	ProvisionerClassName string `json:"provisionerClassName"`
-	// Template describes the generated Bundle that this deployment will manage.
-	Template BundleTemplate `json:"template"`
+	// Source defines the configuration for the underlying Bundle content.
+	Source BundleSource `json:"source"`
 	// Config is provisioner specific configurations
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Config runtime.RawExtension `json:"config,omitempty"`
 }
 
-// BundleTemplate defines the desired state of a Bundle resource
-type BundleTemplate struct {
-	// Standard object's metadata.
-	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
-	// +optional
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-	// Specification of the desired behavior of the Bundle.
-	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
-	Spec BundleSpec `json:"spec"`
-}
-
 // BundleDeploymentStatus defines the observed state of BundleDeployment
 type BundleDeploymentStatus struct {
 	Conditions         []metav1.Condition `json:"conditions,omitempty"`
-	ActiveBundle       string             `json:"activeBundle,omitempty"`
+	ResolvedSource     *BundleSource      `json:"resolvedSource,omitempty"`
+	ContentURL         string             `json:"contentURL,omitempty"`
 	ObservedGeneration int64              `json:"observedGeneration,omitempty"`
 }
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 //+kubebuilder:resource:scope=Cluster,shortName={"bd","bds"}
-//+kubebuilder:printcolumn:name="Active Bundle",type=string,JSONPath=`.status.activeBundle`
 //+kubebuilder:printcolumn:name="Install State",type=string,JSONPath=`.status.conditions[?(.type=="Installed")].reason`
 //+kubebuilder:printcolumn:name=Age,type=date,JSONPath=`.metadata.creationTimestamp`
 //+kubebuilder:printcolumn:name=Provisioner,type=string,JSONPath=`.spec.provisionerClassName`,priority=1
 
-// BundleDeployment is the Schema for the bundledeployments API
+// BundleDeployment is the Schema for the bundleDeployments API
```

Comment:
When we're talking about the API and using the plural resource name, I'd suggest leaving it all lower-case to match the resource name in the API.
```suggestion
// BundleDeployment is the Schema for the bundledeployments API
```